src/main/java/org/truffleruby/parser/ast/ParseNode.java

```diff
@@ -84,6 +84,8 @@ public void setPosition(SourceIndexLength position) {
     }
     
     public abstract <T> T accept(NodeVisitor<T> visitor);
+
+    /** Only for debugging, cast and use getters on specific classes instead. */
     public abstract List<ParseNode> childNodes();
```

Comment:
Could annotate with `@Deprecate` then?

Reply:
That would force to annotate all `childNodes()` deprecated too, otherwise there are warnings.
Maybe we should just remove that method (in another PR) and just reimplement ParseNode#toString() to read fields instead, similar to how it works for Truffle nodes.